src/models/Section/Repository.ts

```diff
@@ -1,29 +1,31 @@
-import { IUpdateSectionModel, SectionType, IFindByEntity } from "./Interfaces";
+import { IUpdateSectionModel, IFindByEntity } from "./Interfaces";
 
-export const SectionRepository = {
-  update: async <Section, Entity>({
-    entityUuidKey,
-    modelClass,
+export abstract class SectionRepository {
+  public async updateSection<Entity>({
     sections,
     entity,
     transaction
-  }: IUpdateSectionModel<Section, Entity>) => {
-    await modelClass.destroy({ where: { [entityUuidKey]: entity.uuid }, transaction });
+  }: IUpdateSectionModel<Entity>) {
+    await this.modelClass().destroy({
+      where: { [this.entityUuidKey()]: entity.uuid },
```

Comment:
entity es un nombre muy poco especifico, puede cumplir cualquier rol... en este caso usaria algo como owner

tal vez algo como `where: this.whereClause(owner)`